examples/ota-provider-app/linux/main.cpp

```diff
@@ -50,31 +52,104 @@ using chip::Messaging::ExchangeManager;
 constexpr chip::EndpointId kOtaProviderEndpoint = 0;
 
 constexpr uint16_t kOptionFilepath             = 'f';
+constexpr uint16_t kOptionOtaImageList         = 'o';
 constexpr uint16_t kOptionQueryImageBehavior   = 'q';
 constexpr uint16_t kOptionDelayedActionTimeSec = 'd';
 
 // Global variables used for passing the CLI arguments to the OTAProviderExample object
-OTAProviderExample::QueryImageBehaviorType gQueryImageBehavior = OTAProviderExample::kRespondWithUpdateAvailable;
-uint32_t gDelayedActionTimeSec                                 = 0;
-const char * gOtaFilepath                                      = nullptr;
+static OTAProviderExample::QueryImageBehaviorType gQueryImageBehavior = OTAProviderExample::kRespondWithUpdateAvailable;
+static uint32_t gDelayedActionTimeSec                                 = 0;
+static const char * gOtaFilepath                                      = nullptr;
+static const char * gOtaImageListFilepath                             = nullptr;
+
+
+// Parses the JSON filepath and extracts DeviceSoftwareVersionModel parameters
+static bool parse_json_file_and_populate_candidates(const char* filepath, std::vector<OTAProviderExample::DeviceSoftwareVersionModel> & candidates)
+{
+    bool ret = false;
+    Json::Value root;
+    Json::CharReaderBuilder builder;
+    JSONCPP_STRING errs;
+    std::ifstream ifs;
+
+    builder["collectComments"] = true; // allow C/C++ type comments in JSON file
+    ifs.open(filepath);
```

Comment:
We should probably do some kind of `if (!ifs.good())` check unless parseFromStream does that for you?

Reply:
Makes sense!